src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/ProductPreferencesFormHandler.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/*
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2018 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Configure\ShopParameters\ProductPreferences;
+
+use PrestaShop\PrestaShop\Adapter\Cache\CacheClearer;
+use PrestaShop\PrestaShop\Core\Form\FormHandler;
+
+/**
+ * Class manages the data manipulated using forms
+ * in "Configure > Shop Parameters > Product Settings" page.
+ */
+class ProductPreferencesFormHandler extends FormHandler
+{
+    /**
+     * @var CacheClearer
+     */
+    private $cacheClearer;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(array $data)
+    {
+        $errors = $this->formDataProvider->setData($data);
```

Comment:
Correct me if im wrong, but it seems that $this->formDataProvider->setData($data); is called in parent::save($data); as well.
It could be like this maybe:
```php
$errors = parent::save($data);

if (empty($errors)) {
    // if logic here
}

return $errors;
```

Reply:
I didn't know if you needed to save once "first" or only once, it's a minor refactoring.

If you think it's worth doing it this way, I update.

Reply:
yes, it's planned but not in this pr: I need this to be merged before friday when the update you suggest can be done during freeze period.

We will do it anyway :+1: